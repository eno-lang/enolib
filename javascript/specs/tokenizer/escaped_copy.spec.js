const { inspectTokenization } = require('./util.js');

const input = `
\`key\` < template

\`\`ke\`y\`\` < template

\`\`\`k\`\`ey\`\`\`    < template

    \`\` \`key\` \`\`    < template

\`key\` < template
`.trim();

describe('Escaped copy tokenization', () => {
  it('performs as specified', () => {
    expect(inspectTokenization(input)).toMatchSnapshot();
  });
});
