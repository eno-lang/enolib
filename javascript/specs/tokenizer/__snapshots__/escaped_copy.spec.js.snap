// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Escaped copy tokenization performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "key": "template",
        "line": 0,
        "parent": [Circular],
        "ranges": Object {
          "elementOperator": Array [
            8,
            9,
          ],
          "key": Array [
            0,
            8,
          ],
          "line": Array [
            0,
            9,
          ],
        },
        "type": Symbol(Field, Fieldset or List),
      },
      Object {
        "key": "key",
        "line": 2,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            17,
            18,
          ],
          "escapeBeginOperator": Array [
            11,
            12,
          ],
          "escapeEndOperator": Array [
            15,
            16,
          ],
          "key": Array [
            12,
            15,
          ],
          "line": Array [
            11,
            27,
          ],
          "template": Array [
            19,
            27,
          ],
        },
        "template": "template",
        "type": Symbol(Field, Fieldset or List),
      },
      Object {
        "key": "ke\`y",
        "line": 4,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            38,
            39,
          ],
          "escapeBeginOperator": Array [
            29,
            31,
          ],
          "escapeEndOperator": Array [
            35,
            37,
          ],
          "key": Array [
            31,
            35,
          ],
          "line": Array [
            29,
            48,
          ],
          "template": Array [
            40,
            48,
          ],
        },
        "template": "template",
        "type": Symbol(Field, Fieldset or List),
      },
      Object {
        "key": "k\`\`ey",
        "line": 6,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            65,
            66,
          ],
          "escapeBeginOperator": Array [
            50,
            53,
          ],
          "escapeEndOperator": Array [
            58,
            61,
          ],
          "key": Array [
            53,
            58,
          ],
          "line": Array [
            50,
            75,
          ],
          "template": Array [
            67,
            75,
          ],
        },
        "template": "template",
        "type": Symbol(Field, Fieldset or List),
      },
      Object {
        "key": "\`key\`",
        "line": 8,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            96,
            97,
          ],
          "escapeBeginOperator": Array [
            81,
            83,
          ],
          "escapeEndOperator": Array [
            90,
            92,
          ],
          "key": Array [
            84,
            89,
          ],
          "line": Array [
            77,
            106,
          ],
          "template": Array [
            98,
            106,
          ],
        },
        "template": "template",
        "type": Symbol(Field, Fieldset or List),
      },
      Object {
        "key": "key",
        "line": 10,
        "parent": [Circular],
        "ranges": Object {
          "copyOperator": Array [
            114,
            115,
          ],
          "escapeBeginOperator": Array [
            108,
            109,
          ],
          "escapeEndOperator": Array [
            112,
            113,
          ],
          "key": Array [
            109,
            112,
          ],
          "line": Array [
            108,
            124,
          ],
          "template": Array [
            116,
            124,
          ],
        },
        "template": "template",
        "type": Symbol(Field, Fieldset or List),
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 11,
  "_meta": Array [],
}
`;
